File: IdeaProjects/Car Parking Lot System/src/models/ParkingTicket.java
package models;

import Interface.Payment;
import Interface.Vehical;

import java.util.Date;

public class ParkingTicket {
    private int ticketNo;
    private Date extryTimestamp;
    private Date exitTimestamp;
    private double amount;
    private boolean status;

    private Vehical vehical;
    private Payment payment;
    private Enterance enterance;
    private Exit exit;
}

File: 09_Design Problems/01_Parking Lot System/Essentials/ParkingTicket.cpp
#include <iostream>
#include "../Interfaces/Vehicle.cpp"
#include "../Essentials/Entrance.cpp"
#include "../Essentials/Exit.cpp"
#include "../Interfaces/Payment.cpp"

using namespace std;

class ParkingTicket {
    int ticketNo;
    time_t EntryTimeStamp;
    time_t ExitTimeStamp;
    double price;
    bool status;

    Vehicle* vehicle;
    IPayment* payment;
    Entrance* entrance;
    Exit* exit;
};